app/controllers/key_pairs_controller.rb

```diff
@@ -38,36 +49,91 @@ def edit
 
   # Given a File object to upload, save it on the file system with
   # association to the user_name given.
-  # Admins are able to save keys under different user names then
-  # their own.
+  # If a String is supplied as the first argument then it's content
+  # is used to create the public key
   # Creates the KEY_STORAGE directory if it does not yet exist
-  def upload_key_file(_file, user_name)
+  def upload_key_file(isFile, _file, time_stamp)
     Dir.mkdir(KEY_STORAGE) unless File.exists?(KEY_STORAGE)
-    uploaded_io = _file
 
-    # todo: check current user to see if they have permission to
-    # use a different user_name then their current one (admins)
+    public_key_content = ''
+
+    if !isFile
+      public_key_content = _file
+    else
+      public_key_content = _file.read
+    end
 
-    File.open(Rails.root.join(KEY_STORAGE, user_name + '.pub'), 'wb') do |f|
-      f.write(uploaded_io.read)
+    File.open(Rails.root.join(KEY_STORAGE, @current_user.user_name + "@" +
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.